cluster/engine.go

```diff
@@ -526,6 +525,10 @@ func (e *Engine) RemoveImage(name string, force bool) ([]types.ImageDelete, erro
 	rmOpts := types.ImageRemoveOptions{force, true}
 	dels, err := e.apiClient.ImageRemove(context.Background(), name, rmOpts)
 	e.CheckConnectionErr(err)
+	if err != nil {
+		return nil, err
```

Comment:
I suspect `ImageRemove` is not atomic. That is, it could have deleted some layers but failed to finish it. In that case, we might still need to refresh images. 

@vdemeester, correct me if I'm wrong. 

```
// ImageRemove removes an image from the docker host.
func (cli *Client) ImageRemove(ctx context.Context, imageID string, options types.ImageRemoveOptions) ([]types.ImageDelete, error) 
```